video/out/gpu_next/context.c

```diff
@@ -146,16 +146,23 @@ struct gpu_ctx *gpu_ctx_create(struct vo *vo, struct gl_video_opts *gl_opts)
     if (ra_is_gl(ctx->ra_ctx->ra)) {
         struct GL *gl = ra_gl_get(ctx->ra_ctx->ra);
         pl_opengl opengl = pl_opengl_create(ctx->pllog,
+# if HAVE_EGL
             pl_opengl_params(
                 .debug = ctx_opts->debug,
                 .allow_software = ctx_opts->allow_sw,
                 .get_proc_addr_ex = (void *) gl->get_fn,
                 .proc_ctx = gl->fn_ctx,
-# if HAVE_EGL
                 .egl_display = eglGetCurrentDisplay(),
                 .egl_context = eglGetCurrentContext(),
-# endif
             )
+# else
+            pl_opengl_params(
+                .debug = ctx_opts->debug,
+                .allow_software = ctx_opts->allow_sw,
+                .get_proc_addr_ex = (void *) gl->get_fn,
+                .proc_ctx = gl->fn_ctx,
+            )
```

Comment:
To remove duplication, you could move init of params to variable and set egl fields after if needed